src/workflow/ArcanistDiffWorkflow.php

```diff
@@ -2804,11 +2807,20 @@ private function pushChangesToStagingArea($id) {
       $ref_list[] = $ref['commit'].':'.$ref['ref'];
     }
 
-    $err = phutil_passthru(
-      'git push %Ls -- %s %Ls',
-      $push_flags,
-      $staging_uri,
-      $ref_list);
+    // phutil_passthru wraps the given string with single quotes
+    // therefore, "GIT_TRACE=1" is not passed through %s in string formatting
+    if ($this->getArgument('trace-git-push'))
```

Comment:
Have you tried set ENV variable before executing this code?
```
if (...) {
  putenv("GIT_TRACE=1");
}
```

If that does not work, update passthru string to include eng variables in the beginning and set variable if condition is met. This will let you avoid code duplication.